pkg/security/ebpf/c/activity_dump.h

```diff
@@ -105,13 +112,81 @@ __attribute__((always_inline)) struct cgroup_tracing_event_t *get_cgroup_tracing
     return evt;
 }
 
+struct bpf_map_def SEC("maps/traced_cgroups_lock") traced_cgroups_lock = {
+    .type = BPF_MAP_TYPE_HASH,
+    .key_size = sizeof(u32),
+    .value_size = sizeof(u32),
+    .max_entries = 1,
+};
+
+__attribute__((always_inline)) bool lock_cgroups_counter() {
+    u32 key = 0;
+    return bpf_map_update_elem(&traced_cgroups_lock, &key, &key, BPF_NOEXIST) == 0;
+}
+
+__attribute__((always_inline)) void unlock_cgroups_counter() {
+    u32 key = 0;
+    bpf_map_delete_elem(&traced_cgroups_lock, &key);
+}
+
+__attribute__((always_inline)) bool reserve_traced_cgroup_spot(char cgroup[CONTAINER_ID_LEN]) {
+    void *already_in = bpf_map_lookup_elem(&traced_cgroups, &cgroup[0]);
+    if (already_in) {
+        return false;
+    }
+
+    if (!lock_cgroups_counter()) {
+        return false;
+    }
+
+    u32 key = 0;
+    struct traced_cgroups_counter_t *counter = bpf_map_lookup_elem(&traced_cgroups_counter, &key);
+    if (!counter) {
+        unlock_cgroups_counter();
+        return false;
+    }
+
+    bool res = false;
+    if (counter->counter < counter->max) {
+        counter->counter++;
+        res = true;
+    }
+
+    unlock_cgroups_counter();
+    return res;
+}
+
+__attribute__((always_inline)) void freeup_traced_cgroup_spot(char cgroup[CONTAINER_ID_LEN]) {
+    if (!lock_cgroups_counter()) {
+        return;
+    }
+
+    bpf_map_delete_elem(&traced_cgroups, &cgroup[0]);
+
+    u32 key = 0;
+    struct traced_cgroups_counter_t *counter = bpf_map_lookup_elem(&traced_cgroups_counter, &key);
+    if (!counter) {
+        unlock_cgroups_counter();
+        return;
+    }
+
+    __sync_fetch_and_add(&counter->counter, -1);
```

Comment:
why do you need to use this atomic function here and not in the `reserve_traced_cgroup_spot` function. It seems they both use the lock stuff ?

Reply:
Oh yes good point, this is a leftover of some of my lock experimentations